arangodb-http/src/main/scala/io/funkode/arangodb/http/ArangoClientHttp.scala

```diff
@@ -123,10 +123,24 @@ class ArangoClientHttp[Encoder[_], Decoder[_]](
   def withConfiguration(newConfig: ArangoConfiguration): ArangoClient[Encoder, Decoder] =
     new ArangoClientHttp[Encoder, Decoder](newConfig, httpClient, token)
 
-  private def parseResponseBody[O: Decoder](response: Response)(using Decoder[ArangoError]): AIO[O] =
+  private def parseResponseBody[O: Decoder](response: Response): AIO[O] =
     for body <-
         if response.status.isError
-        then httpDecoder.decode[ArangoError](response.body).flatMap[Any, ArangoError, O](r => ZIO.fail(r))
+        then
+          response.body.asString
+            .mapError(e =>
+              ArangoError(500, true, "Can't get body of response from Arango: " + e.getMessage, -1)
+            )
+            .flatMap(body =>
+              ZIO.fail(
+                ArangoError(
+                  500,
```

Comment:
Here we need to parse ArangoError to get proper error handling (for example 404 not found error)

Alternatives:
- fast fix -> leave this code but instead of 500 use response.status
- proper fix -> try to do httpDecoder.decode[ArangoError](response.body) and if the parsing fails then use this generic 500 error

Example:
```scala
ZIO.fromEither(httpDecoder.decode[ArangoError](response.body))
          .mapError(decodeError =>
            ArangoError(
              500,
              true,
              s"Error parsing JSON Arango response: " + decodeError + s"\nBody: $body",
              -1
            )

Reply:
Done